### PR TITLE
#1248 - NERFormModal

### LINKED ISSUE #1258: Make an NERModal
### Description

Make a styled, NERModal

### Acceptance Criteria

We have a new NERModal to standardize how all our modals look

### Proposed Solution

Do it

### Mocks

_No response_

### PR DESCRIPTION
**Pull request by RChandler234**
## Changes

I'm trying to make a component for an NERModal, so we can use it for all these finance modals. Lmk if you think the NERFormModal is overkill, I made it cause a lot of our modals that have forms have that bit of code that's the same every time.

I only migrated 3 of the modals to show how we can use this component. I think migrating the rest of the modals would be good tickets for the fall.

## Screenshots

image

## To Do

- [ ] Make a generic NERModal and more specific NERFormModal
- [ ] Migrate rest of modals (maybe, unless we wanna make them into easy tickets)

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [ ] All commits are tagged with the ticket number
- [ ] No linting errors / newline at end of file warnings
- [ ] All code follows repository-configured prettier formatting
- [ ] No merge conflicts
- [ ] All checks passing
- [ ] Screenshots of UI changes (see Screenshots section)
- [ ] Remove any non-applicable sections of this template
- [ ] Assign the PR to yourself
- [ ] No `yarn.lock` changes (unless dependencies have changed)
- [ ] Request reviewers & ping on Slack
- [ ] PR is linked to the ticket (fill in the closes line below)

Closes #1258  (issue #)

### CODE REVIEW

**Comment by RChandler234** [comment]
image
image
image

**Comment by RChandler234** [comment]
image

**Comment by RChandler234** [comment]
image

File: src/frontend/src/components/NERModal.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            color: (theme) => theme.palette.text.primary
+          }}
+        >
+          <CloseIcon />
+        </IconButton>
+      )}
+      <DialogContent sx={{ '&.MuiDialogContent-root': { paddingTop: '20px' } }}>{children}</DialogContent>
+      <DialogActions>
+        <Box sx={{ display: 'flex', flexDirection: 'row' }}>
+          <NERFailButton sx={{ mx: 1, mb: 1 }} form={formId} onClick={onHide}>
```
Comment (Peyton-McKee) [inline src/frontend/src/components/NERModal.tsx:54]: remove the mb from here and just put in the box or something

**Review by Peyton-McKee — APPROVED** [review]
looks good with an optional change doesnt really matter

**Pull request merged**